tokio/src/io/mod.rs

```diff
@@ -108,7 +109,10 @@
 //!         // write a byte to the buffer
 //!         writer.write(&[42u8]).await?;
 //!
-//!     } // the buffer is flushed once writer goes out of scope
+//!         // flush the buffer before it gets out of scope
+//!         writer.flush().await?;
+//!
+//!     } // unless flushed/shutdown the contents of the buffer get discarded on drop
```

Comment:
Shutdown is a noun. It needs a space when it is a verb or adjective.
```suggestion
//!     } // Unless flushed or shut down, the contents of the buffer is discarded on drop.
```